### PR TITLE
Add split argument to required builders and set it default value to "train"

### PR DESCRIPTION
**Pull request by krammnic**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [X] fix a bug
- [X] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.
#1770 
#### Changelog
What are the changes made in this PR?
* Add split argument to required builders and set it default value to "train"

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [X] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [X] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [X] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [x] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [X] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by krammnic** [comment]
@joecummings @RdoubleA Require review

**Review by joecummings — COMMENTED** [review]
This generally looks good, but can you include some output where you attempt to run on a dataset without specifying train and it starts without any errors?

**Comment by krammnic** [comment]
For multimodal:

```
from torchtune.datasets.multimodal import multimodal_chat_dataset


ds = multimodal_chat_dataset(
    model_transform=None,
    source="Lin-Chen/ShareGPT4V",
    split="train",
    name="ShareGPT4V",
    image_dir="/home/user/dataset/",
    image_tag="<image>",
)

```

Output: 
```
README.md: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2.22k/2.22k [00:00<00:00, 10.9MB/s]
(…)egpt4v_instruct_gpt4-vision_cap100k.json: 100%|██████████████████████████████████████████████████████████████████████████████| 134M/134M [00:02<00:00, 56.2MB/s]
Generating train split: 100%|████████████████████████████████████████████████████████████████████████████████████| 102025/102025 [00:02<00:00, 49778.44 examples/s]
```

For instruct:

```
from torchtune.datasets import instruct_dataset

dataset = instruct_dataset(
    tokenizer=None,
    source="json",
    data_files="my_dataset.json",
    column_map={
         "input": "question",
         "output": "answer",
    },
    train_on_input=False,
    packed=False,
)
```

my_dataset.json

```
[
    {
        "question": "What time is it in London?",
        "answer": "It is 10:00 AM in London.",
    }
]
```

Output:

Generating train split: 1 examples [00:00, 122.93 examples/s]

For chat:

```
from torchtune.datasets import chat_dataset

dataset = chat_dataset(
    tokenizer=None,
    source="json",
    data_files="my_dataset.json",
    conversation_column="conversations",
    conversation_style="sharegpt",
    train_on_input=False,
    packed=False
)
```

my_dataset.json

```
[
    {
        "conversations": [
            {
                "from": "human",
                "value": "What time is it in London?",
            },
            {
                "from": "gpt",
                "value": "It is 10:00 AM in London.",
            }
        ]
    }
]
```

Output:

```
Generating train split: 1 examples [00:00, 111.08 examples/s]
```

Should it be covered with test?

**Comment by krammnic** [comment]
@joecummings Has provided an examples, but it definitely should be find cause the same argument was passed in **kwargs of SFTDataset

**Review by RdoubleA — APPROVED** [review]
This is excellent, thank you so much for adding this!
